### DISCUSSION TITLE
LLM Playground

### DISCUSSION THREAD

**Discussion by reza-mohideen** [langfuse/langfuse#1170] [Ideas]
### Describe the feature or potential improvement

I would like the ability to test out prompts within langfuse itself instead of having to switch back and forth between my IDE and langfuse when finalizing a prompt. When adding a prompt to prompt management I should be able to connect to my model and have some control over output like max tokens and temperature. This way I am able to iterate much more quickly and stay within langfuse when creating prompts.

**Comment by marcklingen** [comment]
Agree, ideally we aim for a playground which can replay values inserted within prompt templates to backtest a new prompt iteration quickly on production data. No timeline on this yet but I am very excited about it

**Reply by raulperula** [reply]
This is one of the key features I missed. For most computational linguists, having this playground will facilitate a lot of experimentation to try different prompting methodologies applied to a specific context or use case.
I hope you will add this in the near future.

**Comment by Stormyy** [comment]
Great new feature! Very useful to test new prompts for different models very quickly. Maybe an improvement for the future could be to add the possibility to mock tools by API or manually supplying data so a prompt for a RAG / Agent flow can be tested/experimented with.

**Reply by marcklingen** [reply]
Thought about different options for this already as being able to iterate on a complex prompt would be really useful.

the ideas fall into the following groups
- chaining prompts within prompt playground
- mock tools via api calls to other internal services
- reused prompts within each other
- mock context/inputs via tracing or datasets

**Comment by marcklingen** [comment]
released today during launch week as @Stormyy already realized: https://langfuse.com/changelog/2024-04-23-prompt-playground

**Comment by reza-mohideen** [comment]
Hi @marcklingen, this is amazing to see! Are there plans to support using our own models. Like those we have on AzureOpenAI or AWS bedrock. Asking for privacy reasons as we self host langfuse.

**Reply by marcklingen** [reply]
We used langchain js + the vercel ai sdk as abstractions to build this partially model-agnostic. we plan to extend support to more models but I cannot promise any timeline on this

**Comment by MillionWang0724** [comment]
Hello, I am using a locally deployed service. May I ask if Playground's capabilities will be open-source? I have noticed that you have placed Playground in the/web/src/ee directory. Are you planning to charge for it? If there is no plan, when will it be released to the open source version
@marcklingen

**Comment by ilkersigirci** [comment]
Adding Playground feature to Self-hosted version would be awesome. I hope we get the open source version of it soon
